src/test/java/teammates/test/cases/ui/StudentProfilePictureActionTest.java

```diff
@@ -189,22 +190,85 @@ private void testActionForStudentWithEmptyGoogleId() {
         assertEquals("", result.blobKey);
     }
 
-    protected void testActionWithEmailAndCourseUnauthorisedInstructor() {
+    protected void testActionWithEmailAndCourseUnauthorisedInstructorOrStudent() {
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, StringHelper.encrypt(student.email),
+                Const.ParamsNames.COURSE_ID, StringHelper.encrypt(student.course)
+        };
+        
         ______TS("Failure case: instructor not from same course");
         AccountAttributes unauthInstructor = dataBundle.accounts.get("instructor1OfCourse2");
         gaeSimulation.loginAsInstructor(unauthInstructor.googleId);
 
+        action = getAction(submissionParams);
+        try {
+            action.executeAndPostProcess();
+            signalFailureToDetectException("Unauthorised Access");
+        } catch (UnauthorizedAccessException uae) {
+            assertEquals("User is not in the course that student belongs to", uae.getMessage());
+        }
+        
+        ______TS("Failure case: student not from same course");
+        AccountAttributes unauthStudent = dataBundle.accounts.get("student1InArchivedCourse");
+        gaeSimulation.loginAsStudent(unauthStudent.googleId);
+
+        action = getAction(submissionParams);
+        try {
+            action.executeAndPostProcess();
+            signalFailureToDetectException("Unauthorised Access");
+        } catch (UnauthorizedAccessException uae) {
+            assertEquals("User is not in the course that student belongs to", uae.getMessage());
+        }
+        
+        ______TS("Failure case: student not from same team");
+        
+        StudentAttributes studentFromDifferentTeam = dataBundle.students.get("student5InCourse1");
+        gaeSimulation.loginAsStudent(studentFromDifferentTeam.googleId);
+
+        action = getAction(submissionParams);
+        try {
+            action.executeAndPostProcess();
+            signalFailureToDetectException("Unauthorised Access");
+        } catch (UnauthorizedAccessException uae) {
+            assertEquals("Student does not have enough privileges to view the photo", uae.getMessage());
+        }
+    }
+    
+    protected void testActionWithEmailAndCourseUnauthorisedInstructorOrStudentMasquerade() {
+        gaeSimulation.loginAsAdmin("admin.user");
```

Comment:
I think you're getting it mixed up. The idea is not someone who _has_ access masquerading as someone who _does not have_ access (although it's good to test this too). The more important case for access control purpose is that someone who _does not have_ access masquerading as someone who _has_ access should still be denied. Quoting my previous comment:
> Check if (1) a student from another team, (2) a student from another course, (3) an instructor from another course, trying to masquerade as either one of the above [the instructors of the course and students in the same team] are denied access to the profile picture.